### PR TITLE
Fixed Artboards.

### PR DESCRIPTION
**Pull request by Tod-Rive**
- No more phantom artboards during PIE.
- Artboards in the level reset on PIE begin.

### CODE REVIEW

File: Source/Rive/Private/Rive/RiveTextureObject.cpp [hunk 1]
```diff
@@ -144,7 +151,11 @@ FVector2f URiveTextureObject::GetLocalCoordinatesFromExtents(const FVector2f& In
 
 void URiveTextureObject::Initialize(const FRiveDescriptor& InRiveDescriptor)
 {
-	Artboard = nullptr;
+	if (Artboard == nullptr)
```
Comment (schetle) [inline Source/Rive/Private/Rive/RiveTextureObject.cpp:154]: Let's move the new artboard, and Reinitialization of the artboard into RiveReady.

File: Source/Rive/Public/Rive/RiveTextureObject.h [hunk 2]
```diff
@@ -77,6 +77,7 @@ class RIVE_API URiveTextureObject : public URiveTexture, public FTickableGameObj
 
 #if WITH_EDITOR
 	void EditorBeginPlay();
+	void OnBeginPIE(bool bIsSimulating);
 	void OnEndPIE(bool bIsSimulating);
 	bool bHasBegunPlay = false;
```
Comment (schetle) [inline Source/Rive/Public/Rive/RiveTextureObject.h:82]: We can probably remove bHasBegunPlay at this point. We can also remove the small block of code in Tick that references it, and calls EditorBeginPlay

**Review by schetle — CHANGES_REQUESTED** [review]

File: Source/Rive/Public/Rive/RiveTextureObject.h (hunk 2 above)
Comment (Tod-Rive) [reply Source/Rive/Public/Rive/RiveTextureObject.h]: Done.

File: Source/Rive/Private/Rive/RiveTextureObject.cpp (hunk 1 above)
Comment (Tod-Rive) [reply Source/Rive/Private/Rive/RiveTextureObject.cpp]: Done.

**Review by schetle — APPROVED** [review]

**Pull request merged**